ProcessMaker/Managers/OauthTransportManager.php

```diff
@@ -0,0 +1,178 @@
+                'client_secret' => $this->token->client_secret
+            )
+        );
+        $client->setAuthConfig($authConfig);
+        $client->setAccessToken((array) $this->token);
+        $accessToken = $this->token->access_token;
+
+        if ($client->isAccessTokenExpired()) {
+            $newToken = $client->fetchAccessTokenWithRefreshToken($this->token->refresh_token);
+            $client->setAccessToken($newToken['access_token']);
+            $accessToken = $newToken['access_token'];
+
+            $this->updateEnvVar("EMAIL_CONNECTOR_GMAIL_API_ACCESS_TOKEN{$index}", $accessToken);
+            $this->updateEnvVar("EMAIL_CONNECTOR_GMAIL_API_REFRESH_TOKEN{$index}", $newToken['refresh_token']);
+            $this->updateEnvVar("EMAIL_CONNECTOR_GMAIL_API_ACCESS_TOKEN_EXPIRE_DATE{$index}", $newToken['expires_in']);
+            $this->updateEnvVar("EMAIL_CONNECTOR_GMAIL_API_TOKEN_CREATED{$index}", $newToken['created']);
+        }
+        
+        return $accessToken;
+    }
+
+    public function checkForExpiredOffice365AccessToken($index) 
+    {
+        $now = new DateTime();
+        $now->format('Y-m-d H:i:s');
+        $expireDate = $this->token->expires_in;
+        $accessToken = $this->token->access_token;
+
+        if ($now->format('Y-m-d H:i:s') > $expireDate ) {
+            $accessToken = $this->refreshAccessToken();
+        }
+        return $accessToken;
+    }
+
+    private function refreshAccessToken() 
+    {
+        try {
+            $index = $this->emailServerIndex ? "_{$this->emailServerIndex}" : '';
+            $guzzle = new Client();
+            $url = 'https://login.microsoftonline.com/' . $this->token->tenant_id . '/oauth2/v2.0/token';
+            $newToken = json_decode($guzzle->post($url, [
+                'form_params' => [
+                    'client_id' =>  $this->token->client_id,
+                    'client_secret' => $this->token->client_secret,
+                    'scope' => 'https://outlook.office.com/IMAP.AccessAsUser.All https://outlook.office.com/POP.AccessAsUser.All https://outlook.office.com/SMTP.Send offline_access',
+                    'refresh_token' => $this->token->refresh_token,
+                    'grant_type' => 'refresh_token',
+                ],
+            ])->getBody()->getContents());
+            
+            $now = new DateTime();
+            $expireTime = $now->add(new DateInterval('PT' . $newToken->expires_in . 'S'));
+            $expireDate = $expireTime->format('Y-m-d H:i:s');
+
+            $this->updateEnvVar("EMAIL_CONNECTOR_OFFICE_365_ACCESS_TOKEN{$index}", $newToken->access_token);
+            $this->updateEnvVar("EMAIL_CONNECTOR_OFFICE_365_REFRESH_TOKEN{$index}", $newToken->refresh_token);
+            $this->updateEnvVar("EMAIL_CONNECTOR_OFFICE_365_ACCESS_TOKEN_EXPIRE_DATE{$index}", $expireDate);
+        return $newToken->access_token;
+
+        } catch (Throwable $error) {
+            throw $error;
```

Comment:
as the error is simply re-thrown after the catch,  the try/catch block is unnecesary. Add some error handling (log info, etc.) to keep the try/catch